### ISSUE TITLE
Trilinos / CDash:  Workflow for creating a cdash report w/out network

### ISSUE THREAD

**Issue by jjellio** [trilinos/Trilinos#9073]
## Question
@trilinos/framework @bartlettroscoe 

I'd like to be able to build on a remote machine (w/out SNL/public internet access), then `package up` the CDash information and `report it` from a different machine.

Is that possible?

If that isn't possible, is there documentation on setting up a CDash dashboard for Trilinos (any thing special SNL does?)
I'd like to be able to track configure/build/tests much the same as testing.sandia.gov does.

I've read through the end of the build reference, but I'm not getting a feel for an `offline` or `remote` submission process.
https://docs.trilinos.org/files/TrilinosBuildReference.html

**Comment by bartlettroscoe** [comment]
> Is that possible?

@jjellio, yes.  If you are using `TRIBITS_CTEST_DRIVER()` under the hood (like with the ATDM Trilinos configuration does and what is documented with https://docs.trilinos.org/files/TrilinosBuildReference.html), just run with:

```
export CTEST_DO_SUBMIT=OFF
```

(see [CTEST_DO_SUBMIT](https://menloservice.sandia.gov/http://tribits.org/doc/TribitsDevelopersGuide.html#ctest-do-submit)), copy all of the `*.xml` files created under `<build_dir>/Testing/<date_time/` to the other machine, and then do wput each of these `*.xml` files to the site https://testing.sandia.gov/cdash/submit.php from that machine.  The details of the last submit part may need tweaked but that is basically it.  For more details, we can consult with @zackgalbreath from Kitware.  There are several projects at SNL that submit XML files to CDash without the help of CTest.

Related to my epic SEPW-213 (PLEASE IGNORE)

**Comment by jjellio** [comment]
I'll have to look into it.  The build is currently driven by spack - but spack `generates` and 'ATDM environment` e.g., `load-env.sh something /path/to/my/generated_directory/`

It should be possible to use that `environment.sh` file and drive a build via the `ctest driver`.

I assume I'd have to implement some file that maps the machine + compiler stuff - because none of it is standard. The `ATDM_CONFIG_COMPILER` name is whatever I made up, but it is does work well with `ATDMDevSettings` - and it sets all of the ATDM `hostname` parts so that CDash should work as well.

I'd like to do this for El Cap.

**Comment by jjellio** [comment]
To be clear,

I'd like to post something like
`atdm-redwood-cce-11.0.4-rocm-4.0.1-hipcc-12.0.4-mvapich2_nogpu-release-shared` (that's a pretty close estimate to what the build name really is)

**Comment by jjellio** [comment]
Would like details on the `submit` part.

I'd probably slurp the files to my SRN - VPN connected mac, then post them.

In a perfect world, we'd have a cdash website behind the SRN firewall (to avoid OUO/NDA issues).
If it's not absurdly complicated, could I stand up a `cdash` server in something like `docker`? Then I'd just submit to that IP.

**Comment by jjellio** [comment]
@bartlettroscoe 
Ross, when I click 'TRIBITS_CTEST_DRIVER().'
at
https://docs.trilinos.org/files/TrilinosBuildReference.html#id115
it wants to send me to:
https://tribits.org/doc/TribitsDevelopersGuide.html#tribits-ctest-driver

But I am getting redirected to:
https://menloservice.sandia.gov/http://tribits.org/doc/tribitsdevelopersguide.html#tribits-ctest-driver

I haven't tried it off VPN. I believe menloservice is SNL's url intercept .

SNL is having a temper tantrum that I am using Chrome to access http://tribits.org. wtf


I'll need to workout the SPACK <-> Ctest driver stuff - but that actually doesn't seem that far-fetched.

maybe sort out a docker for CDash as well...

**Comment by bartlettroscoe** [comment]
> I haven't tried it off VPN. I believe menloservice is SNL's url intercept .

@jjellio, it just started that very recently.  I may need to ask why this is happening.  But by starting at: tribits.org, I was able to get to:

* https://menloservice.sandia.gov/http://tribits.org/doc/TribitsDevelopersGuide.html#tribits-ctest-driver

> maybe sort out a docker for CDash as well...

Having a local CDash that you can directly submit to from the client where you run ctest - S would make things much easier.  Also, if you are submitting from a machine with any sensitivity then you don't want to submit to testing.sandia.gov/cdash/ since that is world-readable.  You really need an internal protected CDash site to submit to.

**Comment by bartlettroscoe** [comment]
>In a perfect world, we'd have a cdash website behind the SRN firewall (to avoid OUO/NDA issues).

@jjellio, we have CDash sites behind the SRN.  Contact me offline and I can point you in the right direction.

**Comment by jjellio** [comment]
I don't want to use an CDash local to the machine, because that Machine is behind a vendor's firewall.... getting Trilinos people to see it would be hard.

I'll drop you an email - we can track this on SRN / confluence stuff if you want.

**Comment by bartlettroscoe** [comment]
> But I am getting redirected to:
https://menloservice.sandia.gov/http://tribits.org/doc/tribitsdevelopersguide.html#tribits-ctest-driver

@jjellio, FYI, but we have tracked this down to a defect in menloservice in that it incorrectly lower-cases the filepaths in URLs when it translates them.  I reported this to the powers that be and hopefully they will fix this at some point.  I put in a workaround in commit https://github.com/TriBITSPub/TriBITS/commit/aefad25c77381d3a9bca9359ebbf6b7f84ba7d29 that is used to publish to tribits.org to create lower-case copies of these files and now those lower-case links will work.
